Source/WebCore/xml/XSLTUnicodeSort.cpp

```diff
@@ -101,41 +100,44 @@ void xsltUnicodeSortFunction(xsltTransformContextPtr ctxt, xmlNodePtr *sorts, in
 
     for (j = 0; j < nbsorts; j++) {
         comp = static_cast<xsltStylePreComp*>(sorts[j]->psvi);
-        tempstype[j] = 0;
         if ((comp->stype == NULL) && (comp->has_stype != 0)) {
-            comp->stype =
+            xmlChar* stype =
                 xsltEvalAttrValueTemplate(ctxt, sorts[j], (const xmlChar *) "data-type", XSLT_NAMESPACE);
-            if (comp->stype != NULL) {
-                tempstype[j] = 1;
-                if (xmlStrEqual(comp->stype, (const xmlChar *) "text"))
-                    comp->number = 0;
-                else if (xmlStrEqual(comp->stype, (const xmlChar *) "number"))
-                    comp->number = 1;
+            number[j] = 0;
+            if (stype) {
+                if (xmlStrEqual(stype, reinterpret_cast<const xmlChar*>("text"))) {
+                    // number[j] already zero.
```

Comment:
Maybe worth asserting?

Reply:
No, you're missing the point of the empty block.

The code that would go here would be `number[j] = 0;`, but that's already done above, so this block has nothing to do other than to check for a valid value.  (I know, it's not WebKit style, but again, I didn't want to restructure this too much to make future changes easier to apply.)